src/actions/settings.ts

```diff
@@ -0,0 +1,23 @@
+import { createAction } from 'redux-actions'
+import { State, Settings } from 'types'
+import { Dispatch } from 'redux'
+
+import localForage from 'localforage'
+
+export const saveSettings = createAction<Partial<Settings>>('SAVE_SETTINGS')
+
+export const asyncLoadSettings = () => async (dispatch: Dispatch<Settings>) => {
+  const settings = await localForage.getItem('settings') as Settings
+
+  if (settings) {
+    return dispatch(saveSettings(settings))
+  }
+}
+
+export const asyncSaveSettings = (payload: Partial<Settings>) =>
+  async (dispatch: Dispatch<Settings>, getState: () => State) => {
+    const action = dispatch(saveSettings(payload))
```

Comment:
why not just return the `dispatch` or `localForage` save directly ? why cache action?

Reply:
Traditionally `dispatch(action)` returns that same action, thunks return `Promise<action>`. By analogy I return the `action`. Doesn't really matter